packages/ember-toucan-form/src/components/toucan-form.ts

```diff
@@ -0,0 +1,39 @@
+import Component from '@glimmer/component';
+
+import TextareaFieldComponent from '../-private/textarea-field';
+
+import type { WithBoundArgs } from '@glint/template';
+import type { HeadlessFormComponentSignature } from 'ember-headless-form/components/headless-form';
+
+// TODO: Should we export this from ember-headless-form and use it here instead? This is probably fine?
+type UserData = object;
+
+type HeadlessFormArguments<
+  DATA extends UserData,
+  SUBMISSION_VALUE
+> = HeadlessFormComponentSignature<DATA, SUBMISSION_VALUE>['Args'];
+
+export interface ToucanFormComponentSignature<
+  DATA extends UserData,
+  SUBMISSION_VALUE
+> {
+  Element: HTMLFormElement;
+  Args: HeadlessFormArguments<DATA, SUBMISSION_VALUE>;
+  Blocks: {
+    default: [
+      {
+        Textarea?: WithBoundArgs<typeof TextareaFieldComponent, 'form'>;
+        // TODO: Need to figure out a type for this
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        Field: any;
```

Comment:
@simonihmig kind of similar to above - we want a type for this, so maybe exposing a type from headless-form we can use, define it here ourselves, or something else?

Reply:
Yeah, it's basically solved when the other problem is solved: when we have the type for `form`, then this is as simple as `form['Field']`...